RapidReact/src/main/java/frc/robot/subsystem/ClimberSubsystem.java

```diff
@@ -1,114 +1,503 @@
+          nextState = ClimbState.ExtendPartial;
+        }
+        break;
+      case ExtendFull:
+        if (isClimberAtSetpoint(fullExtendPosition)) {
+          nextState = ClimbState.Retract;
+          retractInit();
+        } else {
+          nextState = ClimbState.ExtendFull;
+        }
+        break;
+      case Retract:
+        if (isClimberAtSetpoint(fullRetractPosition)) {
+          nextState = ClimbState.Idle;
+          idleInit();
+        } else {
+          nextState = ClimbState.Retract;
+        }
+        break;
+    }
+
+    // climberRight.follow(climberLeft, FollowerType.AuxOutput1);
+
+    currentClimbState = nextState;
+  }
 
   @Override
   public void disable() {
-    climber.set(0);
+    climberLeft.set(0);
+    climberRight.set(0);
   }
 
   public void toggleClimberEnabled() { // uses 2 PS button
     climberEnabled = !climberEnabled;
-    
-    if (climberEnabled){
+
+    if (climberEnabled) {
       climbState.log(LogLevel.GENERAL, "climberEnabled");
-    } else{
+    } else {
       climbState.log(LogLevel.GENERAL, "climberDisabled");
     }
   }
 
-  public void fixedHookToggler() { //uses R1 button
-    if (config.enablePneumatics) {
-      fixedHookToggleState = !fixedHookToggleState;
-      if (!fixedHookToggleState) {
-        fixedHookSolenoid.set(Value.kForward);
-        hookToggleState.log(LogLevel.GENERAL, "fixedHookTime");
-      } else {
-        fixedHookSolenoid.set(Value.kReverse);
-        hookToggleState.log(LogLevel.GENERAL, "fixedHookTime");
-      }
-    }
-  }
+  public void manualElevatorExtend() { //uses up button
+    if (autoClimb) return;
 
-  public void elevatorExtend() { //uses up button
-    climber.set(ControlMode.PercentOutput, climbOutput.currentValue());
+    // TODO: LIMIT SWITCHES https://docs.ctre-phoenix.com/en/stable/ch13_MC.html#limit-switches
+    // TODO: you should have the joystick/ button move the motion magic setpoint, not the motor in PWM mode
+    climberLeft.set(ControlMode.PercentOutput, climbOutput.currentValue());
+    // climberRight.follow(climberLeft, FollowerType.AuxOutput1);
+    climberRight.set(ControlMode.PercentOutput, climbOutput.currentValue());
 
     climbState.log(LogLevel.GENERAL, "elevatorExtend");
   }
 
-  public void elevatorRetract() { //uses down button
-    climber.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+  public void manualElevatorRetract() { //uses down button
+    if (autoClimb) return;
+
+    // TODO: LIMIT SWITCHES https://docs.ctre-phoenix.com/en/stable/ch13_MC.html#limit-switches
+    // TODO: you should have the joystick/ button move the motion magic setpoint, not the motor in PWM mode
+    climberLeft.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+    // climberRight.follow(climberLeft, FollowerType.AuxOutput1);
+    climberRight.set(ControlMode.PercentOutput, -climbOutput.currentValue());
 
     climbState.log(LogLevel.GENERAL, "elevatorRetract");
   }
 
   public void elevatorStop() {
```

Comment:
Why do you have both this function and disable()? They seem to do the same thing.

Reply:
Simply stopping the motors vs a full subsystem disable should be logically separated I think, calling "disable()" when we just want to stop the motors doesn't seem right